bin/crystal

```diff
@@ -94,27 +94,22 @@ _canonicalize_file_path() {
 
 ##############################################################################
 
-# Based on http://stackoverflow.com/a/2108540/2199687
+# Based on http://stackoverflow.com/q/370047/641451
 remove_path_item() {
-  local path item work
+  local path item
 
   path="$1"
   item="$2"
 
-  work=":$path:"
-  work="${work/:$item:/:}"
-  work="${work%:}"
-  work="${work#:}"
-
-  echo "$work"
+  echo -n $path | awk -v RS=: -v ORS=: '$0 != "'$item'"' | sed 's/:$//'
 }
 
 ##############################################################################
 
 __has_colors() {
   local num_colors=$(tput colors 2>/dev/null)
 
-  if [ "$num_colors" -gt 2 ]; then
+  if [ "0$num_colors" -gt 2 ]; then
```

Comment:
It may be a bit too hacky, yes.

The idea is that, if `tput` is not available (like when building in Docker), then `$num_colors` would be empty - and the `[` will fail. This way, if `$num_colors` is empty, the string gets to be `"0"`, while if it's not empty, `"256"` turns into `"0256"` - and still works.

I can change that to be a different test - ie, if it's empty, assume no color; else test for `$num_colors` - but I should get the dust off my shell scripting book (?)